docs/guide.adoc

```diff
@@ -497,7 +498,7 @@ points := [1, 2, 3]
 
 sum := 6
 
-points_str := points -> map' fn p: int -> str do
```

Comment:
You do know this is still allowed?

Reply:
Yes, but I don't see the point of being inconsistent. I'd say we remove that syntax, but it's not hurting anyone.

Reply:
Removing it is gonna be tricky - but I like the optionality of it. It might also be helpfull if we want to allow functions definitions over multiple lines. I can also see it being used as a stylistic approach when inlining with a return type:
```
add_one :: fn a -> int do a + 1 end
```
This reads clearer than:
```
add_one :: fn a -> int a + 1 end
```
If you want to explicitly write the return type.